core/block/editor/template.go

```diff
@@ -68,28 +92,17 @@ func (t *Template) getTypeKeyById(typeId string) (domain.TypeKey, error) {
 	return domain.TypeKey(uniqueKey.InternalKey()), nil
 }
 
-// GetNewPageState returns state that can be safely used to create the new document
-// it has not localDetails set
-func (t *Template) GetNewPageState(name string) (st *state.State, err error) {
-	st = t.NewState().Copy()
+func (t *Template) updateTypeKey(st *state.State) error {
 	objectTypeID := pbtypes.GetString(st.Details(), bundle.RelationKeyTargetObjectType.String())
 	if objectTypeID != "" {
 		typeKey, err := t.getTypeKeyById(objectTypeID)
 		if err != nil {
-			return nil, fmt.Errorf("get target object type %s: %w", objectTypeID, err)
+			return fmt.Errorf("get target object type %s: %w", objectTypeID, err)
 		}
 		st.SetObjectTypeKey(typeKey)
+		return nil
 	}
-	st.RemoveDetail(bundle.RelationKeyTargetObjectType.String(), bundle.RelationKeyTemplateIsBundled.String())
-	st.SetDetail(bundle.RelationKeySourceObject.String(), pbtypes.String(t.Id()))
-	// clean-up local details from the template state
-	st.SetLocalDetails(nil)
-
-	if name != "" {
-		st.SetDetail(bundle.RelationKeyName.String(), pbtypes.String(name))
-		if title := st.Get(template.TitleBlockId); title != nil {
-			title.Model().GetText().Text = ""
-		}
-	}
-	return
+	updatedTypeKeys := slice.Remove(t.ObjectTypeKeys(), bundle.TypeKeyTemplate)
```

Comment:
hmmm, I noticed that slice.Remove mutates original slice, it could lead to terrible consequences 

could you please implement immutable Remove function in our slice lib? And rename `slice.Remove` to `slice.RemoveMut`?

Reply:
you could use `golang.org/x/exp/slices` under the hood

Reply:
Done! Implemented new Remove func using Copy function for exp/slices